lib/puppet/util/fips.rb

```diff
@@ -0,0 +1,28 @@
+require 'puppet/util/platform'
+# Method to determine whether where are running on a FIPs enabled host
+
+module Puppet::Util::FIPS
+  include Puppet::Util::Platform
+
+  def self.fips_enabled?
+    # We currently only plan to support FIPS on RHEL7
+    # so we try to eliminate anything un-supported using
+    # what we have available in lieu of not being able to specifically
+    # check for it.
+    if Puppet::Util::Platform.windows?
+      false
```

Comment:
I think you want `return false if Puppet::Util::Platform.windows?` otherwise we fall through and continue along.